test/executor.test.js

```diff
@@ -906,6 +906,62 @@ describe('executor', function() {
     });
   });
 
+  describe('when booting with lazy connect', function() {
+    var sampleInstruction = someInstructions({
+      dataSources: {
+        lazyConnector: {
+          connector: 'testLazyConnect',
+          name: 'lazyConnector',
+        },
+      },
+    });
+    var connectTriggered = true;
+
+    beforeEach(function() {
+      // Reset boolean value to true
+      connectTriggered = true;
+      app.connector('testLazyConnect', {
+        initialize: function(dataSource, callback) {
+          if (dataSource.settings.lazyConnect) {
+            connectTriggered = false;
+          }
+        },
+      });
+    });
+
+    it('should trigger connect with ENV undefined', function(done) {
+      delete process.env.LB_LAZYCONNECT_DATASOURCES;
+      boot.execute(app, sampleInstruction, function() {
+        expect(connectTriggered).to.be.true;
```

Comment:
[Beware of libraries that assert on property access](https://github.com/moll/js-must#asserting-on-property-access). We should start using dirty-chai, but that's out of scope of this patch. As a workaround, use this please:

```js
expect(connectTriggered).to.equal(true);
```

See also https://github.com/strongloop/eslint-config-loopback/pull/6